### PR TITLE
sdjournal: Break the loop if error happens during Write().

### PR DESCRIPTION
**Pull request by yifan-gu**
This fixes the endless loop issue reported by @alban .

### CODE REVIEW

**Comment by philips** [comment]
LGTM.

**Comment by philips** [comment]
cc @jonboulle

File: sdjournal/read.go [hunk 1]
```diff
@@ -161,7 +161,9 @@ process:
 			return ErrExpired
 		default:
 			if c > 0 {
-				writer.Write(msg[:c])
+				if _, err := writer.Write(msg[:c]); err != nil {
```
Comment (alban) [inline sdjournal/read.go]: With `:=`, the inner "err" variable masks the outer "err" variable and the function will not return any error. Is it intended and/or correct?

Comment (yifan-gu) [inline sdjournal/read.go]: good catch

**Comment by alban** [comment]
LGTM now

**Comment by flyingmutant** [comment]
FYI, this PR breaks the build with

```
./read.go:164: multiple-value writer.Write() in single-value context
```

It looks like the CI does not cover sdjournal, unfortunately.

**Comment by jonboulle** [comment]
Yeah, just noticed that :-(

**Comment by yifan-gu** [comment]
oops, my bad

**Comment by alban** [comment]
Fixed via #175 / #177

**Pull request merged**